src/commands/compile/compileCommand.ts

```diff
@@ -0,0 +1,122 @@
+import { CommandExample, ReturnCode } from '../../types/command'
+import { TargetCommand } from '../../types/command/targetCommand'
+import { displayError } from '../../utils'
+import { compile } from './compile'
+import path from 'path'
+import {
+  getDestinationJobPath,
+  getDestinationServicePath
+} from './internal/getDestinationPath'
+import { compileSingleFile } from './compileSingleFile'
+
+enum CompileSubCommand {
+  Job = 'job',
+  Service = 'service',
+  Identify = 'identify'
+}
+
+const syntax = 'compile [subCommand]'
+const aliases = ['c']
+const usage = 'Usage: sasjs compile [subCommand] [options]'
+const description =
+  'Compiles jobs and services in the project by inlining all dependencies and adds init and term programs as configured in the specified target.'
+const examples: CommandExample[] = [
+  {
+    command: 'sasjs compile -t myTarget',
+    description: ''
+  },
+  {
+    command: 'sasjs c -t myTarget',
+    description: ''
+  }
+]
+
+const subCommandParseOptions = {
+  source: { type: 'string', alias: 's', demandOption: true },
+  output: { type: 'string', alias: 'o' }
+}
+
+export class CompileCommand extends TargetCommand {
+  constructor(args: string[]) {
+    let parseOptions = {}
+    const subCommand = args[3]
+    if (
+      [
+        CompileSubCommand.Job,
+        CompileSubCommand.Identify,
+        CompileSubCommand.Service
+      ].includes(subCommand as CompileSubCommand)
+    ) {
+      parseOptions = subCommandParseOptions
+    }
+    super(args, { parseOptions, syntax, usage, description, examples, aliases })
+  }
+
+  public get source(): string {
+    return this.parsed.source as string
+  }
+
+  public get output(): string {
+    const value = this.parsed.output as string
+    let sourcefilePathParts = this.source.split(path.sep)
+    sourcefilePathParts.splice(-1, 1)
+    const sourceFolderPath = sourcefilePathParts.join(path.sep)
+    const leafFolderName = sourceFolderPath.split(path.sep).pop() as string
+    return value
+      ? path.isAbsolute(value)
```

Comment:
this ternary operator is a bit hard to read

Reply:
fixed.

Reply:
not fixed!

Reply:
I have further refactored this piece of code to: 
`
    let outputPath: string
    if (value) {
      const internal = [value, `${this.subCommand}s`, leafFolderName]
      outputPath = getAbsolutePath(internal.join(path.sep), process.currentDir)
    } else {
      outputPath =
        this.subCommand === 'job'
          ? getDestinationJobPath(sourceFolderPath)
          : getDestinationServicePath(sourceFolderPath)
    }
`